### PR TITLE
recording: record build packages installed in the snap

### PR DESCRIPTION
**Pull request by elopio**

### CODE REVIEW

**Comment by elopio** [comment]
Requires #1293

File: integration_tests/test_snapcraft_recording.py [hunk 1]
```diff
@@ -18,29 +18,235 @@
...
+    )
+
+    def test_prime_records_packages_version(self):
+        """Test the recorded snapcraft.yaml for a snap with packages
+
+        This snap declares all the packages that it requires, there are
+        no additional dependencies. The packages specify their version.
+        """
+        self.copy_project_to_cwd(
+            '{}-without-dependencies'.format(self.packages_type))
```
Comment (kalikiana) [inline integration_tests/test_snapcraft_recording.py]: You forgot to stage build-packages-without-dependencies.

Comment (elopio) [inline integration_tests/test_snapcraft_recording.py]: thanks for catching it, again.

File: integration_tests/test_snapcraft_recording.py [hunk 2]
```diff
@@ -53,12 +57,196 @@ def test_prime_records_snapcraft_yaml(self):
...
+        with open(recorded_yaml_path) as recorded_yaml_file:
+            recorded_yaml = yaml.load(recorded_yaml_file)
+
+        self.assertEqual(recorded_yaml, source_yaml)
+
+    def test_prime_without_packages_version(self):
+        """Test the recorded snapcraft.yaml for a snap with packages
+
+        This snap declares all the packages that it requires, there are
+        no additional depenencies. The packages dont' specify their
```
Comment (sergiusens) [inline integration_tests/test_snapcraft_recording.py]: dependencies and do not

File: snapcraft/internal/repo/_deb.py [hunk 3]
```diff
@@ -229,7 +229,13 @@ def install_build_packages(cls, package_names):
...
     @classmethod
     def get_installed_build_packages(cls, package_names):
-        unique_packages = set(package_names)
+        # It's important to preserve the order of packages to record on the
+        # state the same value received from the snapcraft.yaml
+        seen = set()
+        unique_packages = [
+            package for package in package_names if not
+            (package in seen or seen.add(package))
+        ]
```
Comment (sergiusens) [inline snapcraft/internal/repo/_deb.py]: good catch

**Pull request merged**